app/controllers/v0/sessions_controller.rb

```diff
@@ -35,11 +35,16 @@ def saml_callback
 
     def persist_session_and_user!
       @session = Session.new(user_attributes.slice(:uuid))
-      @current_user = User.find(@session.uuid) || create_new_user
+      @current_user = User.find(@session.uuid)
+      @current_user = saml_user if @current_user.nil? || up_level?
```

Comment:
This is a concern of mine as well.   Down-leveling is not something we will support so the logic of:
````ruby
if @current_user.loa < saml_user.loa
  # dont even bother comparing attributes
```
is sound.  The grayness comes when the two loa's are equal.  In that case I think we might want to compare attributes and re-create the user - which potentially means making another MVI call. (see sessions_controller:87)